docs/developer/solutions/archive.rst

```diff
@@ -0,0 +1,177 @@
+Solution archive guidelines
+===========================
+
+To provide a predictable interface with packaged Solutions, MetalK8s expects a
+few criteria to be respected, described below.
+
+
+Archive format
+--------------
+
+Solution archives must use the `ISO-9660:1988`_ format, including `Rock Ridge`_
+and Joliet_ directory records. The character encoding must be UTF-8_. The
+conformance level is expected to be at most 3, meaning:
+
+- Directory identifiers may not exceed 31 characters (bytes) in length
+- File name + ``'.'`` + file name extension may not exceed 30 characters
+  (bytes) in length
+- Files are allowed to consist of multiple sections
+
+The generated archive should specify a volume ID, set to
+``{project_name} {version}``.
+
+.. todo::
+
+   Clarify whether Joliet/Rock Ridge records supersede the conformance level
+   w.r.t. filename lengths
+
+
+.. _`ISO-9660:1988`: https://www.iso.org/obp/ui/#iso:std:iso:9660:ed-1:v1:en
+.. _`Rock Ridge`: https://en.wikipedia.org/wiki/Rock_Ridge
+.. _Joliet: https://en.wikipedia.org/wiki/Joliet_(file_system)
+.. _UTF-8: https://tools.ietf.org/html/rfc3629
+
+Here is an example invocation of the common Unix mkisofs_ tool to generate such
+archive::
+
+   mkisofs
+       -output my_solution.iso
+       -R  # (or "-rock" if available)
+       -J  # (or "-joliet" if available)
+       -joliet-long
+       -l  # (or "-full-iso9660-filenames" if available)
+       -V 'MySolution 1.0.0'  # (or "-volid" if available)
+       -gid 0
+       -uid 0
+       -iso-level 3
+       -input-charset utf-8
+       -output-charset utf-8
+       my_solution_root/
+
+.. _mkisofs: https://linux.die.net/man/8/mkisofs
+
+.. todo::
+
+   Consider if overriding the source files UID/GID to 0 is necessary
+
+
+File hierarchy
+--------------
+
+Here is the file tree expected by MetalK8s to exist in each Solution archive::
+
+   .
+   ├── images
+   │   └── some_image_name
+   │       └── 1.0.1
+   │           ├── <layer_digest>
+   │           ├── manifest.json
+   │           └── version
+   ├── operator
+   │   ├── crd
+   │   │   └── some_crd_name.yaml
+   │   └── deployment.yaml
+   ├── product.txt
+   └── ui
+       └── deployment.yaml
+
+.. _solution-archive-product-info:
+
+Product information
+-------------------
+
+General product information about the packaged Solution must be stored in the
+``product.txt`` file, stored at the archive root.
+
+It must respect the following format::
+
+   NAME=Example
+   VERSION=1.0.0-dev
+   REQUIRE_METALK8S=">=2.0"
+
```

Comment:
One thing must be added here: some kind of 'format version', such that we can change the ISO layout over time, and dispatch accordingly during ISO import (in the MetalK8s code).